pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -3752,6 +3756,7 @@ private CompletableFuture<Void> updateSubscriptionsDispatcherRateLimiter() {
     }
 
     protected CompletableFuture<Void> initTopicPolicy() {
+        registerTopicPolicyListener();
```

Comment:
Why do you modify this? This should be not related with this issue.

Reply:
Referring to the above flow diagram, method `onUpdate` and method `this.updateTopicPolicyByNamespacePolicy(policies)` run in parallel.

And it's possible that method `registerTopicPolicyListener()->onUpdate()->checkReplication()` is called before method `this.updateTopicPolicyByNamespacePolicy(policies)`.

Therefore, move method `registerTopicPolicyListener()` to execute after method `this.updateTopicPolicyByNamespacePolicy(policies)`.

Here are the steps to reproduce it:
https://github.com/apache/pulsar/issues/21653#issuecomment-1842962452

Reply:
Thanks. Initially, I thought it was more than just this problem and what really works is the above check 
`if (CollectionUtils.isEmpty(configuredClusters)) `.

https://github.com/apache/pulsar/blob/757723e4c03b2aa2e5203e90fdb9ecb255e18a3b/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java#L636-L640

However, it seems that when `startCheckReplicationPolicies` is executed completely, the policy has been successfully updated.